sdk/python/tests/integration/registration/test_universal_registry.py

```diff
@@ -149,7 +149,9 @@ def pg_registry():
 
     registry_config = RegistryConfig(
         registry_type="sql",
-        path=f"postgresql://{POSTGRES_USER}:{POSTGRES_PASSWORD}@{container_host}:{container_port}/{POSTGRES_DB}",
+        # The `path` must include `+psycopg` in order for `sqlalchemy.create_engine()`
+        # to understand that we are using psycopg3.
+        path=f"postgresql+psycopg://{POSTGRES_USER}:{POSTGRES_PASSWORD}@{container_host}:{container_port}/{POSTGRES_DB}",
```

Comment:
Yes :( More context can be found [here](https://docs.sqlalchemy.org/en/20/dialects/postgresql.html#module-sqlalchemy.dialects.postgresql.psycopg) btw!

Reply:
This is indeed unfortunate. Let's plan on drafting comms about this then.

Reply:
A (somewhat hacky) alternative is to add a `pydantic` [field_validator](https://docs.pydantic.dev/latest/api/functional_validators/#pydantic.functional_validators.field_validator) to the `RegistryConfig` class that replaces `postgresql` with `postgresql+psycopg` in the path if `registry_type` equals `sql` and `path` starts with `postgresql`, but not with `postgresql+psycopg`.

Reply:
A (somewhat hacky) alternative is to add a `pydantic` [field_validator](https://docs.pydantic.dev/latest/api/functional_validators/#pydantic.functional_validators.field_validator) to the `RegistryConfig` class that replaces `postgresql` with `postgresql+psycopg` in the path if `registry_type` equals `sql` and `path` starts with `postgresql`, but not with `postgresql+psycopg`.

What are your thoughts on this @tokoko @franciscojavierarceo ?

Reply:
I think that makes sense, but we should also take into account that someone out there might still want to use `psycopg2` behind the registry. So I think the validator should check that only `psycopg3` is installed first and add `+psycopg` to the url with an appropriate warning there as well.

Reply:
> I think that makes sense, but we should also take into account that someone out there might still want to use psycopg2 behind the registry.

Is this still a possible scenario after this PR gets merged? IIUC, all usage of `psycopg2` in `feast` will be replaced by `psycopg`. So `psycopg2` will no longer be a transitive dependency when installing `feast` with extras `postgres`.

Reply:
> there's no reason for the user can't keep using psycopg2

@tokoko But would a user really care that `feast` uses specifically `psycopg2` for the registry instead of the newer and better `psycopg`? 

I'd imagine users to care primarily about using `postgres` for their registry (opposed to some other type of registry), but not really care about whether `feast` is leveraging `psycopg2` or `psycopg` under the hood for its Postgres database operations.

> So I think the validator should check that only psycopg3 is installed first and add +psycopg to the url with an appropriate warning there as well.

This is still something that could be added of course. Just curious if this is really necessary.

Reply:
It's not important, I'm probably being pedantic here. I was thinking of a case when some connection param works with only `psycopg2`, that might be confusing.

But since there's a way to explicitly specify `psycopg2` if a user really wants too (`postgresql+psycopg2`), let's just throw a warning if only `postgresql` is specified and silently change to `postgresql+psycopg`.

Reply:
Thanks for your suggestions! 

I just pushed another commit implementing the suggestion you made. Can you perhaps take another look? :) @tokoko

Reply:
This looks good, thanks